src/components/forms/components/submission-fields.js

```diff
@@ -552,7 +565,7 @@ class ArrayField extends React.Component{
         const schema = propSchema.items || {};
         const values = propValue || [];
         const valuesToRender = _.map( values.length === 0 ? [null] : values , function(v,i){ return [v, schema, i]; });
-        const showAddButton = !isValueNull(values[valuesToRender.length - 1]);
+        const showAddButton = !propSchema.maxItems && !isValueNull(values[valuesToRender.length - 1]);
```

Comment:
Similarly to previous comment, would consider if maxItems can ever be validly 0, because if so, then `!0` will evaluate to `true`. Alternatively could test `typeof propSchema.maxItems === "number"` to determine presence (or lack) of the propSchema.maxItems even if it is 0. In this case it makes sense that maxItems can never be below 1 so would say is fine to leave as is.